src/codecatalyst/auth.ts

```diff
@@ -69,6 +69,14 @@ export class CodeCatalystAuthenticationProvider {
         return this.secondaryAuth.isUsingSavedConnection
     }
 
+    /** True if a CC Builder ID connection is known by the Toolkit*/
+    public async builderIdExists(allConnections = () => this.auth.listConnections()): Promise<boolean> {
+        const predicate = ((conn?: Connection) => {
+            return isBuilderIdConnection(conn) && isValidCodeCatalystConnection(conn)
+        })
+        return (await allConnections()).some(predicate)
```

Comment:
Instead of continuing to duplicate stuff in a spread-out fashion, it would help discovery if `auth/utilts.ts` had a single function that took a `kind='plain'|'codecatalyst'|'codewhisperer'` parameter.

Or something like that. If that introduces a circular dependency, create a new file `src/auth/utils2.ts`.

Reply:
Do we want to move context specific code (eg: CodeWhisperer Identity Center, CodeCatalyst Builder ID) outside of their dedicated auth classes? I'd assume this would make discovery harder since you would have to know to look in `src/auth/utils.ts` for a specific CodeWhisperer functionality, instead of in the existing `src/codewhisperer/util/authUtil.ts` module.

Reply:
> outside of their dedicated auth classes? ...   would make discovery harder since you would have to know to look in src/auth/utils.ts for a specific CodeWhisperer functionality, instead of src/codewhisperer/util/authUtil.ts module.

That is always the case for any generic functionality: generic functionality is found in generic modules, I don't see a discovery problem there. 

The functionality in this PR is generalizable. The special-case auth modules are an accident, not something we want to grow.

Reply:
Also regarding the duplicate test code, these tests are more-so testing the `predicate` of each method instead of the `some` since each implementation has a slightly different predicate.

Reply:
> The special-case auth modules are an accident, not something we want to grow.

I see, will update.